packages/teams-js/test/public/pages.spec.ts

```diff
@@ -210,42 +335,917 @@ describe('AppSDK-TeamsAPIs', () => {
+        lastViewUnixEpochTime: null,
+        entityId: 'MockEntityId',
+        channelId: 'MockChannelId',
+        channelName: 'MockChannelName',
+        channelIsFavorite: true,
+        teamId: 'MockTeamId',
+        teamName: 'MockTeamName',
+        teamIsFavorite: true,
+        groupId: 'MockGroupID',
+        url: 'http://some-valid-content-url.com',
+        websiteUrl: 'http://some-valid-website-url.com',
+      };
+      it('pages.tabs.navigateToTab should not allow calls before initialization', async () => {
+        await expect(() => pages.tabs.navigateToTab(null)).rejects.toThrowError(
+          'The library has not yet been initialized',
+        );
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        it(`pages.tabs.navigateToTab should register the navigateToTab action when initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          pages.tabs.navigateToTab(tabInstance);
+          const navigateToTabMsg = utils.findMessageByFunc('navigateToTab');
+          expect(navigateToTabMsg).not.toBeNull();
+          expect(navigateToTabMsg.args[0]).toBe(tabInstance);
+        });
+        it(`pages.tabs.navigateToTab should register the navigateToTab action when initialized with ${context} context - success case`, async () => {
+          await utils.initializeWithContext(context);
+          pages.tabs.navigateToTab(null);
+          const onComplete = getGenericOnCompleteHandler();
+          onComplete(true);
+          const navigateToTabMsg = utils.findMessageByFunc('navigateToTab');
+          expect(navigateToTabMsg).not.toBeNull();
+          expect(navigateToTabMsg.args[0]).toBe(null);
+        });
+      });
+    });
+
+    describe('Testing pages.tabs.getTabInstances function', () => {
+      let expectedTabInstanceParameters: TabInstanceParameters = {
+        favoriteChannelsOnly: true,
+        favoriteTeamsOnly: true,
+      };
+      it('pages.tabs.getTabInstances should not allow calls before initialization', async () => {
+        await expect(() => pages.tabs.getTabInstances()).rejects.toThrowError(
+          'The library has not yet been initialized',
+        );
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        it(`pages.tabs.getTabInstances should successfully getTabInstance when no parameters are passed and initialized with ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          const promise = pages.tabs.getTabInstances();
+          let message = utils.findMessageByFunc('getTabInstances');
+
+          utils.respondToMessage(message, expectedTabInstanceParameters);
+          expect(message).not.toBeNull();
+          expect(promise).resolves.toBe(expectedTabInstanceParameters);
+        });
+
+        it(`pages.tabs.getTabInstances should be undefined getTabInstance when parameters are passed and  initialized with ${context} context`, async () => {
```

Comment:
The test says it should be "undefined getTabInstance", but I don't see that as the expected value? (Unless I'm missing something.) Also there is an extra space between "and" and "initialized"

Reply:
Fixed in the latest commit